app/components/Wallet/SuiWallet/useNBTCBalance.tsx

```diff
@@ -0,0 +1,52 @@
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+import { useSuiClient, useCurrentAccount } from "@mysten/dapp-kit";
+import type { CoinBalance } from "@mysten/sui/client";
+
+const endsWithNBTC = (input: string): boolean => {
+	const regex = /NBTC$/;
+	return regex.test(input);
+};
+
+export const useNBTCBalance = () => {
+	const suiClient = useSuiClient();
+	const account = useCurrentAccount();
+	const queryClient = useQueryClient();
+
+	const fetchBalance = async (): Promise<CoinBalance | undefined> => {
+		if (!account) {
+			return undefined;
+		}
+		const coins = await suiClient.getAllBalances({
+			owner: account.address,
+		});
```

Comment:
I think we should be able to query single coin balance. @sczembor could you confirm?